### ISSUE TITLE
premium gift payment fail: segfault at 10000 ip 00007f11199aedfe sp 00007ffeb40a42e0 error 4 in libglib-2.0.so.0.7400.4

### ISSUE THREAD

**Issue by powerman** [telegramdesktop/tdesktop#25646]
### Steps to reproduce

1. Try to send Premium Gift to some contact
2. Fill form and click "Pay"

### Expected behaviour

Payment should be processed.

### Actual behaviour

Got empty window, and these lines in kernel log:

```
2022-12-31_11:30:53.13048 kern.info: telegram-deskto[13764]: segfault at 10000 ip 00007f11199aedfe sp 00007ffeb40a42e0 error 4 in libglib-2.0.so.0.7400.4[7f1119956000+92000]
2022-12-31_11:30:53.13051 kern.info: Code: 48 8b 0b 44 8d 70 ff 48 89 c5 4d 89 f7 49 c1 e7 04 4c 01 f9 4c 8b 21 4d 85 e4 0f 84 ec 00 00 00 49 8b 44 24 08 48 85 c0 74 52 <48> 8b 10 49 89 54 24 08 49 89 c4 48 8b 41 08 48 85 c0 75 2e 66 66
```
![изображение](https://user-images.githubusercontent.com/1354301/210135203-d134a99e-424b-4534-b4f1-4d2929a041d9.png)

### Operating system

Gentoo Linux

### Version of Telegram Desktop

4.4.1

### Installation source

Static binary from official website

**Comment by powerman** [comment]
Sometimes it looks different: screenshot same as above with extra `(null)` message on top and this in kernel log:

```
2022-12-31_11:41:44.85847 kern.info: traps: telegram-deskto[532] general protection fault ip:7f56d1035dfe sp:7fff2b7d50e0 error:0 in libglib-2.0.so.0.7400.4[7f56d0fdd000+92000]
```

**Comment by powerman** [comment]
Just built telegram from source (it's a Gentoo, after all) - this way it works.

**Comment by ilya-fedin** [comment]
What desktop environment do you use?

**Comment by powerman** [comment]
No real DE, just Fluxbox. But I've both KDE and Gnome apps/libs installed.

**Comment by ilya-fedin** [comment]
Can you get the trace of the coredump (probably with `coredumpctl gdb`)?

**Comment by powerman** [comment]
I already made the payment, and as this issue triggered by clicking "Pay" button - how I can trigger it without actually made one more extra payment?
Also I need more detailed instructions how to get the coredump - telegram itself didn't crash, and I don't see coredumpctl tool on my system. I can run telegram under gdb, but I need to know which commands to enter in gdb.

**Comment by ilya-fedin** [comment]
Webview acts in a separate process. coredumpctl is a part of systemd, if you don't have systemd, I sadly don't know how you can get the coredump, probably you have to refer to Google or your distro community for the help.

With systemd, after `coredumpctl gdb` you need only to enter `bt full`.

**Comment by ilya-fedin** [comment]
> how I can trigger it without actually made one more extra payment?

I don't think you can sadly.

**Comment by powerman** [comment]
No, I don't have systemd. Probably payment can be blocked by using my bank's app and configuring card limits. So, if you'll find out how to get the trace without systemd - let me know.

**Comment by ilya-fedin** [comment]
After you click pay, it likely redirects to your bank's page and crashes there and it's not possible to reach your bank page without paying something. It sounds then like a duplicate of #25541.

**Comment by powerman** [comment]
When it crash it doesn't make any redirect (i.e. it crash before that) - no new tabs are opened in my browser. When it won't crash - yes, it does open new tab in browser to process payment.

**Comment by ilya-fedin** [comment]
the browser is inside the payment window

**Comment by powerman** [comment]
> the browser is inside the payment window

Usually when I buy something using telegram bots - yes, no tabs opened in my browser and payment has processed completely in telegram. But this time it has opened page in real browser (maybe it's because I made payment using new card - i.e. card details was added before clicking "Pay").

**Comment by ilya-fedin** [comment]
I don't quite understand why you fixed on the browser, you think a redirect can't happen without opening tabs in browser or what?

**Comment by powerman** [comment]
I'm not sure is this important. I just point out there are two different payment flows, one involve opening new tab in real browser, other uses browser embedded in telegram.

**Comment by ilya-fedin** [comment]
It likely first redirects you to some bank's page and then that bank page does a JavaScript call to open another bank's page in a new tab (what leads the request to go to the browser)
